functorch/functorch/csrc/BatchRulesLinearAlgebra.cpp

```diff
@@ -293,6 +293,42 @@ oneOutput matrix_exp_batch_rule(const Tensor& self, c10::optional<int64_t> self_
   return std::make_tuple(at::matrix_exp(self_), 0);
 }
 
+ fourOutputs solve_ex_batch_rule(
+    const Tensor& A, optional<int64_t> A_bdim,
+    const Tensor& B, optional<int64_t> B_bdim,
+    bool left, bool check_errors) {
+  auto batch_size = get_bdim_size2(A, A_bdim, B, B_bdim);
+  const auto A_logical_rank = rankWithoutBatchDim(A, A_bdim);
+  const auto B_logical_rank = rankWithoutBatchDim(B, B_bdim);
+  const auto max_logical_rank = std::max(A_logical_rank, B_logical_rank);
+
+  TORCH_CHECK(A_logical_rank >= 2,
+            "linalg.solve: The input tensor A must have at least 2 dimensions.");
+
+  int b_logical_rank = max_logical_rank;
+  if (A_logical_rank > B_logical_rank) {  // vector case: B was a vector or batched vector
```

Comment:
Is this correct? The [docs](https://pytorch.org/docs/stable/generated/torch.linalg.solve.html) say that A could be of shape (L, N, N) while B can be of shape (N, K).  B is still a matrix in this case.

Though, I'm confused, because it sounds like solve is ambiguous if A has shape (N, N, N) and B has shape (N, N) - is B a single matrix or a batch of vectors or does it just not matter?

Reply:
From docs:
> If A has shape (*, n, n) and B has shape (*, n) (a batch of vectors)...
> Otherwise, if A has shape (*, n, n) and B has shape (n,) or (n, k) ...

It looks like from testing that it takes the rules in the order listed (i.e. if A has shape (N, N, N) and B has shape (N, N), it will treat the B as a batch of vectors). But agree that it's a little ambiguous

Reply:
ahh the documentation is correct, thanks for pointing that out